lightning/src/ln/channel.rs

```diff
@@ -3974,9 +3974,13 @@ impl<Signer: Sign> Channel<Signer> {
 
 	/// Adds a pending outbound HTLC to this channel, note that you probably want
 	/// send_htlc_and_commit instead cause you'll want both messages at once.
-	/// This returns an option instead of a pure UpdateAddHTLC as we may be in a state where we are
-	/// waiting on the remote peer to send us a revoke_and_ack during which time we cannot add new
-	/// HTLCs on the wire or we wouldn't be able to determine what they actually ACK'ed.
+	/// This returns an option instead of a pure UpdateAddHTLC as we may be in a state where we
+	/// cannot add HTLCs on the wire.
+	/// In cases where we're waiting on the remote peer to send us a revoke_and_ack, if we did, we
+	/// wouldn't be able to determine what they actually ACK'ed when we do receive the
+	/// revoke_and_ack.
+	/// In cases where we're marked MonitorUpdateFailed, we cannot commit to a new state as we may
+	/// not yet have sent the previous commitment update messages and will need to regenerate them.
 	/// You MUST call send_commitment prior to any other calls on this Channel
```

Comment:
Suggestion to make this clearer:

"You MUST call `send_commitment` prior to calling any other methods on this channel."